test/validation/api/ipsec/ipsec_test_out.c

```diff
@@ -731,30 +745,52 @@ static void test_esp_out_in(struct cipher_param *cipher,
 		return;
 
 	if (flags->display_algo)
-		printf("\n    %s (keylen %d) %s (keylen %d) ",
+		printf("\n        %s (keylen %d) %s (keylen %d) ",
 		       cipher->name, cipher_keylen, auth->name, auth_keylen);
 
 	test_out_in_common(flags, cipher->algo, cipher->key,
 			   auth->algo, auth->key,
 			   cipher->key_extra, auth->key_extra);
 }
 
+static int iterate_flags(ipsec_test_flags *flags, uint32_t *iterator)
+{
+next:
+	if (*iterator >= 0x8)
+		return 0;
+
+	flags->v6           = *iterator & 0x4;
+	flags->tunnel       = *iterator & 0x2;
+	flags->tunnel_is_v6 = *iterator & 0x1;
+	*iterator += 1;
+
+	if (!flags->tunnel && flags->tunnel_is_v6)
+		goto next;
```

Comment:
Why not `while` or `for`? `goto`s in ODP I believe are mainly used for error/exit jumps.